### PR TITLE
Fetch schema to validate json

### PR DESCRIPTION
**Pull request by samely**
Fix https://github.com/mapbox/HecateJS/issues/37

Add `ajv` module to validate features according to the JSON schema before importing data into Hecate.

@mapbox/search

### CODE REVIEW

File: lib/import.js [hunk 1]
```diff
@@ -124,7 +125,8 @@ class Import {
 
                 // validate the geojson file before uploading starts
                 const geojsonErrs = validateGeojson(filepath, {
-                    ignoreRHR: options.ignoreRHR
+                    ignoreRHR: options.ignoreRHR,
+                    schema: getSchema()
```
Comment (ingalls) [inline lib/import.js]: You'll need to move this above geojson Errs. Right now this won't return synchronously and will result in the schema being null (or a function) every time.

**Comment by samely** [comment]
Currently, I'm unable to get the schema.

if I run `HECATE_USERNAME=x HECATE_PASSWORD=y  ./cli.js schema get` I get an error

```
{"statusCode":406,"headers":{"server":"GitHub.com","date":"Wed, 13 Nov 2019 18:42:14 GMT","content-type":"application/json","transfer-encoding":"chunked","connection":"close","status":"406 Not Acceptable"...}
```

I'm able to get the schema if I pass a token so we might want to prompt for this information.

cc./ @ingalls @miccolis

**Comment by ingalls** [comment]
This is now starting & cachine the schema as expected, however upon attempting to validate a feature against the schema it runs into a max callstack error

```
schema $id ignored http://json-schema.org/draft-07/schema#
/home/null/Development/HecateJS/node_modules/ajv/lib/compile/resolve.js:239
  traverse(schema, {allKeys: true}, function(sch, jsonPtr, rootSchema, parentJsonPtr, parentKeyword, parentSchema, keyIndex) {
                                            ^

RangeError: Maximum call stack size exceeded
    at /home/null/Development/HecateJS/node_modules/ajv/lib/compile/resolve.js:239:45
    at _traverse (/home/null/Development/HecateJS/node_modules/json-schema-traverse/index.js:65:5)

```

File: lib/import.js [hunk 2]
```diff
@@ -113,41 +120,65 @@ class Import {
...
-                const filepath = path.resolve(options.input);
-                rl = new readLineSync(filepath);
-
                 // validate the geojson file before uploading starts
-                const geojsonErrs = validateGeojson(filepath, {
-                    ignoreRHR: options.ignoreRHR
-                });
-                if (geojsonErrs.length) return cb(new Error(`Invalid GeoJSON: \n${geojsonErrs}`));
-            }
+                new Schema(self.api).get({}, (err, schema) => {
```
Comment (mattciferri) [inline lib/import.js]: Outside of the scope of this PR, but does Schema (and its sibling HecateJs modules) need to be a class if we rarely leverage the reference to the instance? Seems like `api` can just be another param to a bunch of static methods.

File: util/validateGeojson.js [hunk 3]
```diff
@@ -1,85 +1,148 @@
...
+    if (opts.schema) {
+        schema = ajv.compile(opts.schema);
     }
 
-    // Validate each feature
-    function validateFeature(line) {
-        const feature = rewind(JSON.parse(line));
-        linenumber++;
-        let errors = [];
+    return transform(1, (feat, cb) => {
```
Comment (mattciferri) [inline util/validateGeojson.js]: If the parallelism is 1, do we need to use the parallel library?

File: lib/import.js (hunk 2 above)
Comment (ingalls) [reply lib/import.js]: @mattciferri At the moment the library is designed as both a cli and a library, as a library this makes sense, but for the cli, these sections need to be in a standardized format to allow the API to be build automatically.

File: util/validateGeojson.js (hunk 3 above)
Comment (ingalls) [reply util/validateGeojson.js]: @mattciferri It makes implementing a transform stream much easier, and we can experiement with increasing this in the future

**Pull request merged**